int8[] adjency_matrix
---
bool success
